pkg/tools/grpcutils/listen_and_serve.go

```diff
@@ -24,28 +24,25 @@ import (
 
 	"google.golang.org/grpc"
 
-	"github.com/networkservicemesh/sdk/pkg/tools/errctx"
 	"github.com/networkservicemesh/sdk/pkg/tools/log"
 )
 
 const (
 	unixScheme = "unix"
 )
 
-// ListenAndServe listens on address with server.  Returns a context which will be canceled in the event that
-// server.Serve(listener) returns an error.  The resulting error can then be retrieve from the returned context with.
-// grpcutils.Err(ctx)
-func ListenAndServe(ctx context.Context, address *url.URL, server *grpc.Server) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	ctx = errctx.WithErr(ctx)
+// ListenAndServe listens on address with server.  Returns an chan err  which will
+// receive an error and then be closed in the event that server.Serve(listener) returns an error.
+func ListenAndServe(ctx context.Context, address *url.URL, server *grpc.Server) chan error {
```

Comment:
Let's return read-only error channel to prevent closing by the consumer.

Reply:
Done! :)